client-common/src/tendermint/rpc_client/sync_rpc_client.rs

```diff
@@ -310,3 +320,20 @@ impl Client for SyncRpcClient {
             .collect()
     }
 }
+
+impl Drop for SyncRpcClient {
+    fn drop(&mut self) {
+        if self.disconnect_on_drop {
```

Comment:
Instead of having a boolean flag, you can check following condition:
```
if Arc::strong_count(&self.runtime) == 1 {
    // Close connection
}
```

Reply:
ok, that's better solution.
i'll check it